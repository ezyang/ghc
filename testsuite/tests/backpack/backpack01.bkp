package p where
    includes: base
    exposed-modules: P
    required-signatures: Hole
    source-dir: "p"

package q where
    includes: base,
              p (Hole as Data.String, P)
    exposed-modules: Q A
    reexported-modules: A as B, Data.String as X
    source-dir: "q"

executable foo2 where
    includes: q (X, Q), base, p (Hole as X, P)
    main-is: foo

executable foo1 where
    includes: q, base, p (Hole as Data.String, P)
    main-is: foo
