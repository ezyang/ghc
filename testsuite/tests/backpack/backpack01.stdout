
BUILDING package p [p_0pGVquHcc3Y514DjCMqrDd]:
[1 of 2] Compiling Hole[sig of Data.String] ( p/Hole.hsig, nothing )
[2 of 2] Compiling P                ( p/P.hs, p/out_p_0pGVquHcc3Y514DjCMqrDd/P.o )

BUILDING package q [q_6JosentJQf4K0ocJRv0iGz]:
[1 of 2] Compiling A                ( q/A.hs, q/A.o )
[2 of 2] Compiling Q                ( q/Q.hs, q/Q.o )

BUILDING executable foo:
[1 of 1] Compiling Main             ( foo.hs, out_foo2/Main.o )
Linking foo2 ...

BUILDING executable foo:
[1 of 1] Compiling Main             ( foo.hs, out_foo1/Main.o )
Linking foo1 ...
